packages/amplify-category-auth/src/provider-utils/awscloudformation/utils/generate-auth-trigger-template.ts

```diff
@@ -61,9 +69,15 @@ export async function generateNestedAuthTriggerTemplate(context: $TSContext, cat
   const cfnFileName = 'auth-trigger-cloudformation-template.json';
   const targetDir = path.join(pathManager.getBackendDirPath(), category, request.resourceName!);
   const authTriggerCfnFilePath = path.join(targetDir, cfnFileName);
-  const { authTriggerConnections } = request;
+  const { authTriggerConnections, permissions } = request;
   if (authTriggerConnections) {
-    const cfnObject = await createCustomResourceforAuthTrigger(context, JSON.parse(authTriggerConnections));
+    const cfnObject = await createCustomResourceforAuthTrigger(
+      context,
+      JSON.parse(authTriggerConnections),
+      permissions!.map(i => JSONUtilities.parse(i)),
```

Comment:
Using `!` is for sure? it can never be `undefined`? Alsp why parse the `permissions` array per member, why not just once the whole array?

Reply:
Removed !.
 Parsing is like this as data is sent in this format previously from auth walkthrough